src/Poisson.jl

```diff
@@ -71,7 +71,26 @@ end
     return s
 end
 
-residual!(p::Poisson) = @inside p.r[I] = p.z[I]-mult(I,p.L,p.D,p.x)
+"""
+    residual!(p::Poisson)
+
+Computes the resiual `r = z-Ax` and corrects it such that
+`r = 0` if `iD==0` which ensures local satisfiability
+    and 
+`sum(r) = 0` which ensures global satisfiability.
+
+The global correction is done by adjusting all points uniformly, 
+minimizing the local effect. Other approaches are possible.
+
+Note: These corrections mean `x` is not strictly solving `Ax=z`, but
+without the corrections, no solution exists.
+"""
+function residual!(p::Poisson) 
+    @inside p.r[I] = ifelse(p.iD[I]==0,0,p.z[I]-mult(I,p.L,p.D,p.x))
```

Comment:
Shouldn't the `ifelse` be converted for floating point types? Something like:
```
    @inside p.r[I] = ifelse(p.iD[I]≈zeroT,zeroT,p.z[I]-mult(I,p.L,p.D,p.x))
```
And then we can define `zeroT` in `util.jl` according to the `T` type, and export it. I could push a PR for this separately if you'd like. Would be nice to also clean this up in other places.

Reply:
Sure. But why not just using `zero(T)`? I doubt the performance impact will be enough for me to measure on my (noisy) laptop.

Reply:
Maybe we could use ternary operator here? cuz `ifelse(cond, resTrue, resFalse)` will evaluate both `resTrue` and `resFalse` before entering the decision process, while `cond ? resTrue :  resFalse` will only evaluate the necessary one.

Reply:
This is what benchmarks are good for. According to the documentation `ifelse` avoids branch conditions. Which _sounds_ fast...

Reply:
I will setup the jelly benchmark and we see :)

Reply:
This change shouldn't reduce performance on the TGV either (as long as I haven't messed something up). It replaces all the σᵥ code with one `ifelse`.